presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1305,6 +1307,13 @@ else if (sourceNode instanceof SampleNode) {
                     return new PhysicalOperation(operatorFactory, outputMappings, context, source);
                 }
             }
+            catch (PrestoException e) {
+                if (e.getErrorCode() == COMPILER_ERROR.toErrorCode() && getExceptionRootCause(e) instanceof MethodTooLargeException) {
```

Comment:
This comparison should use `equals()` instead of `==` since we don't want to rely on the identity of the `ErrorCode` object. (even though they are singletons today and that is likely to be true in the future, it's not enforced or specified anywhere)

Reply:
Done